New package: iamxiaojianzheng.jd-gui version 1.6.8

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #154838

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/154862)

Dragon1573 committed:
New package: iamxiaojianzheng.jd-gui version 1.6.8

Signed-off-by: 是蓝兔哟～ <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
This package requires a _Java Development Kit_ (version 1.8 or newer) to run properly. So I introduce `Microsoft.OpenJDK.21` here.

Other JDKs are accepted, so maybe `Dependencies` of this package should be adjust.

Dragon1573 commented:
Installation process is verified in Windows Sandbox, and proved to be valid.

Could moderators please provide the error logs? 🙏🏼

stephengillie commented:
Automatic Validation ended with:

> 2024-05-23 08:52:01.989 [CLI ] No suitable installer found for manifest Microsoft.OpenJDK.21 with version 21.0.3.9

(Automated response - build 881.)

Dragon1573 committed:
Update (installer.yaml): Add Scope of the package

- Related #152555
- "Microsoft.OpenJDK.21" install in machine-scope by default, the scope config should add to this package too

Dragon1573 commented on manifests/i/iamxiaojianzheng/jd-gui/1.6.8/iamxiaojianzheng.jd-gui.installer.yaml:
```diff
@@ -0,0 +1,24 @@
+
+PackageIdentifier: iamxiaojianzheng.jd-gui
+PackageVersion: 1.6.8
+Scope: machine
```
`Scope` key has added to `installer.yaml`, but it still can't pass the pipeline. 😕

stephengillie commented:
Automatic Validation ended with:

> 2024-05-23 18:57:41.091 [CLI ] No suitable installer found for manifest Microsoft.OpenJDK.21 with version 21.0.3.9

(Automated response - build 881.)

Dragon1573 commented:
```powershell
.\Tools\SandboxTest.ps1 .\manifests\i\iamxiaojianzheng\jd-gui\1.6.8\ -WinGetOptions '--scope machine'
```

![image](https://github.com/microsoft/winget-pkgs/assets/49941141/0620fcf7-33d2-4754-a47f-df2299283446)
![image](https://github.com/microsoft/winget-pkgs/assets/49941141/a07b64a0-ab39-493e-9115-fc176ac33872)

Dragon1573 commented:
Is this a blocking pull request because of #152555?

stephengillie commented:
@wingetbot run

stephengillie commented:
Automatic Validation ended with:

> 2024-05-28 15:43:33.159 [CLI ] No suitable installer found for manifest Microsoft.OpenJDK.21 with version 21.0.3.9

(Automated response - build 881.)

Dragon1573 commented:
> Automatic Validation ended with:
> 
> > 2024-05-28 15:43:33.159 [CLI ] No suitable installer found for manifest Microsoft.OpenJDK.21 with version 21.0.3.9
> 
> (Automated response - build 881.)

I'll try switching its dependency to another JDK/JRE distribution.

Dragon1573 committed:
Switch architecture: x86 ==> x64

- Since most of JDKs / JREs do NOT have a x64 build, try switching this package to x64.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

stephengillie commented:
Verify: 
![image](https://github.com/microsoft/winget-pkgs/assets/114952565/b4338a7d-b80d-405f-bf0e-1587bcafd1a8)
